lib/index.js

```diff
@@ -22,7 +22,10 @@ function ytdl(link, options) {
 
   getInfo(link, options, function(err, info) {
     if (err) {
-      stream.emit('error', err);
+      stream.emit('error', {
+        error: err,
+        info: info
+      });
```

Comment:
the "error" event should emit an error object. Does this work for you? Because I suspect that if there is an error here, then `info` won't be set to anything. Would be better if there "info" is emitted a few lines down, would be moved up and placed before `downloadFromInfoCallback()`